File: src/main/java/com/github/zxh/classpy/classfile/bytecode/InstructionU1.java
package com.github.zxh.classpy.classfile.bytecode;

import com.github.zxh.classpy.classfile.ClassReader;
import com.github.zxh.classpy.classfile.U1;

/**
 *
 * @author zxh
 */
public class InstructionU1 extends Instruction {

    public InstructionU1(Opcode opcode, int pc) {
        super(opcode, pc);
    }

    @Override
    protected void readOperands(ClassReader reader) {
        U1 operand = reader.readU1();
        setDesc(getDesc() + " " + operand.getDesc());
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/bytecode/InstructionU1.kt
package com.github.zxh.classpy.classfile.bytecode

import com.github.zxh.classpy.classfile.constant.ConstantPool

/**
 * The instruction whose operand is U1.
 */
class InstructionU1(opcode: Opcode, pc: Int) : Instruction(opcode, pc) {

    init {
        u1("opcode")
        u1("operand")
    }

    override fun afterRead(cp: ConstantPool) {
        val operand = super.get("operand")
        desc = "${opcode.name} ${operand.desc}"
    }

}
